### ISSUE TITLE
Status Devs Meeting 5 Agenda

### ISSUE THREAD

**Issue by oskarth** [status-im/pm#5]
Status Devs Meeting 5 Agenda
Meeting Date/Time: Monday 2018-10-01 at 12:00 UTC (14:00 CEST)
Meeting Duration 1.5 hours
YouTube Live Stream Link https://www.youtube.com/c/Statusim/live
Livepeer Live Stream Link TBD

Notes (WIP): https://hackmd.io/WwyyOKHdQOajqzjzgzHqnw

Agenda
Infra updates (infra, p2p, security, testing)
Client updates (mobile, desktop)
Research updates
Product updates (chat, wallet, dapp)
Specific ideas update
Please provide comments to add or correct agenda topics.

**Comment by cammellos** [comment]
* How to share contact lists across multiple devices?

**Comment by oskarth** [comment]
@cammellos Could you please provide more context, possible additional relevant reading and goals of discussion? This will ensure a high quality discussion and best use of time during the actual meeting.

As an example, see:
```
Light clients: LES and ULC.
Context: Currently there's a bit of a disarray w.r.t these efforts - it isn't clear who or what team is owning it or what our deliverables are. Having 10% of users use LES/ULC in app is a p0/p0 OKR, so we need to be clear about what we are doing here, both on a short and longer time scale.
Goal: Discuss approach, reach consensus on way forward, and decide who is leading what focus.

Reading:

status-im/status-go#1025
https://raw.githubusercontent.com/status-im/status-react/develop/doc/decisions/0003-geth-node.md
status-im/ideas#295
```

**Comment by oskarth** [comment]
Last time we had 10 agenda items that were all added at the very last minute. To avoid this situation, let's set a deadline of adding items to the agenda at the latest this week (Friday, possibly) Sunday. Items that are added after that, i.e. on Monday, will not be included. (We can still have an open session for random stuff that comes up, but this won't necessarily be touched on due to time constraints).

In an ideal world, we have 3-5 high impact and high quality items to discuss, with proper context and goals. This ensures the best use of time for all participants, and that we come out of the call with a better collective understanding of things, as well as more precise decisions to follow up on.

If someone strongly disagrees with the above approach, please reply here.

**Comment by jeluard** [comment]
# Chat message bridge

3rd parties want to send message to specific chats using external processes. Simplest way to do that is to offer an HTTP bridge that can be used by anyone.
How do we add PN support to this?

@flexsurfer wrote about it here: https://discuss.status.im/t/whisper-api-as-http-service/470/6

and here:

```
while whisper api is an experiment, and there is no simple solution for services outside status send whisper messages , i see next workflow,
1) dapps ask permission for sending notifications to user (user share his token)
2) dapps server sends notification using FCM SDK and user's token
3) user receives notification in status
4) clicks on it, status opens with some popup message from dapp, something like "CDB is too low in   MakerDAO" and buttons "Cancel' and "Open MakerDAO" ,
5) open MakerDAO dapp if user clicked Open
```

Is that our best option? How about security concerns?

**Comment by adambabik** [comment]
P2P & Infra
========

## Docs and Status Node

First docs how to run Status Node has been prepared. We have a contact to folks running DappNode and Meshbox projects. We want to integrate Status Node with these systems.

## Fork fork fork

We are working towards moving Whisper to a separate package in status-go repository or separate repository. This is because we simply have too many patches and we need to introduce even more after we started working on rate-limiting. 

We also want to remove patches from status-go and just maintain our fork of go-ethereum. This will allow us to use status-go as a library in other projects, for example https://github.com/status-im/statusd-bots.

## Rethinking MailServer v2

Two weeks ago we explored log-based messaging and decetralized storage for MailServers. A few concerns appeared regarding log-based messaging and offloading storing messages on the client though. 

This week I'd like to describe a slightly modified approach which is about providing open source tools to create something called Global Whisper Archives. This approach uses decentralized storage and the idea of forking.

**Comment by PombeirP** [comment]
Regarding Chat message bridge, I think it is reasonable to pass the FCM token to the Dapp server if the user agrees to it. We have some leeway on the client side to detect spam and could notify the user that some service is flooding us with notifications (before even showing them) and potentially allow him to replace his FCM token (currently though, that would mean he'd stop receiving PNs from existing contacts, unless we somehow broadcast the new FCM token to existing contacts).
One important thing to keep in mind is that PN v2 work includes abstracting away the FCM token. You'd send a message through a Whisper secure channel to a contact's mail server to have it send it a push notification to your contact.

**Comment by andytudhope** [comment]
## Documentation Update

We have switched our static site generator, are waiting on a design and to implement a global header, and require one person from each team to agree to be the `documentation janitor` so that we can keep everything ship-shape across the org. We have also set up a proper staging and prod site, so that people can push directly to the `develop` branch in our docs repo, see their changes live, have it reviewed easily, and then merged into `master`, which serves to docs.status.im.

**Comment by corpetty** [comment]
ENS smart contract security audit with Sigma Prime was completed and we are going over their report and collaboratively working on the issues they found.  Once completed, we will release the report to the public.  Hint, we have awesome solidity developers.

**Comment by mandrigin** [comment]
@PombeirP are we though? especially on iOS?

**Comment by PombeirP** [comment]
@mandrigin We aren't filtering yet, but I know we can do it on Android (and I believe on iOS too, but I'm less familiar with the platform).

**Comment by oskarth** [comment]
See https://github.com/status-im/pm/issues/6
